packages/dds/tree/src/shared-tree/sharedTree.ts

```diff
@@ -458,9 +468,9 @@ export class SharedTreeFork implements ISharedTreeFork {
 
 	public fork(): ISharedTreeFork {
 		const anchors = new AnchorSet();
-		const branch = this.branch.fork(anchors);
 		const storedSchema = this.storedSchema.clone();
 		const forest = this.forest.clone(storedSchema, anchors);
+		const branch = this.branch.fork(anchors, () => new ForestRepairDataStore(() => forest));
```

Comment:
Might be nice to make a `repairDataStoreFactoryFromForest` function and put that in the same file as `ForestRepairDataStore` so you don't have to repeat this pattern as much.